### PR TITLE
Add some infrastructure for event logging, including debug events.

### PR DESCRIPTION
**Pull request by graydon**
This is just some preliminary work to have an event buffer and allow writing both protocol-level (TBD) and diagnostic / debug-log (currently useful!) entries to it.

### CODE REVIEW

**Comment by graydon** [comment]
(self-approving since I want it in place for the meeting I'm having with @jayz22 in 5 minutes and also I am going to keep iterating on it this afternoon so will address comments as provided here)

File: stellar-contract-env-common/src/env.rs [hunk 1]
```diff
@@ -64,11 +64,15 @@ macro_rules! call_macro_with_all_host_functions {
...
                 {"_", fn log_value(v:RawVal) -> RawVal }
+
                 /// Get the binary contractID of the contract which invoked the
                 /// running contract. Traps if the running contract was not
                 /// invoked by a contract.
                 {"0", fn get_invoking_contract() -> Object }
                 {"1", fn obj_cmp(a:RawVal, b:RawVal) -> i64 }
+
+                {"2", fn contract_event(v:RawVal) -> RawVal }
+                {"3", fn system_event(v:RawVal) -> RawVal }
```
Comment (leighmcculloch) [inline stellar-contract-env-common/src/env.rs:75]: My understanding from the CAP was that system_event should not be callable, and so it shouldn't be a host fn. It was described as an "internal" fn only.
```
/// Internal host function. Same as $system_event but will use ContractEventType::SYSTEM instead
func $system_event(param $val i64) (result i64)
```
Ref: https://github.com/stellar/stellar-protocol/blob/15d3738523bd577abd711aca2d1582f625d815fc/core/cap-0056.md#host-functions

**Review by leighmcculloch — COMMENTED** [review]
fyi I believe the Horizon team have asked for these to be called logs, not events. They were called logs, then renamed to events, and I think folks want it to be called logs. Ref: https://discord.com/channels/897514728459468821/989220928384806932 cc @sisuresh

**Comment by jonjove** [comment]
I'm a bit confused. Are we actually going in this direction? How does this relate to the conversation in https://github.com/stellar/rs-stellar-contract-env/issues/205#issuecomment-1183553911?

**Pull request merged**